We encountered some issues while trying to upgrade o1js to version 1.5.0. 

Here's a log of the issues for reference:
```
> @zkoracle/utils@0.0.20 build
> tsc

node_modules/o1js/dist/node/lib/mina/actions/offchain-state.d.ts:114:32 - error TS1139: Type parameter declaration expected.

114 declare function OffchainState<const Config extends {
                                   ~~~~~

node_modules/o1js/dist/node/lib/mina/actions/offchain-state.d.ts:114:45 - error TS1005: ',' expected.

114 declare function OffchainState<const Config extends {
                                                ~~~~~~~

node_modules/o1js/dist/node/lib/mina/actions/offchain-state.d.ts:115:9 - error TS1005: ']' expected.

115     [key: string]: OffchainStateKind;
            ~

node_modules/o1js/dist/node/lib/mina/actions/offchain-state.d.ts:115:17 - error TS1005: ',' expected.

115     [key: string]: OffchainStateKind;
                    ~

node_modules/o1js/dist/node/lib/mina/actions/offchain-state.d.ts:115:18 - error TS1180: Property destructuring pattern expected.

115     [key: string]: OffchainStateKind;
                     ~

node_modules/o1js/dist/node/lib/mina/actions/offchain-state.d.ts:115:37 - error TS1005: ',' expected.

115     [key: string]: OffchainStateKind;
                                        ~

node_modules/o1js/dist/node/lib/mina/actions/offchain-state.d.ts:116:1 - error TS1128: Declaration or statement expected.

116 }>(config: Config, options?: {
    ~

node_modules/o1js/dist/node/lib/mina/actions/offchain-state.d.ts:116:2 - error TS1109: Expression expected.

116 }>(config: Config, options?: {
     ~

node_modules/o1js/dist/node/lib/mina/actions/offchain-state.d.ts:116:10 - error TS1005: ')' expected.

116 }>(config: Config, options?: {
             ~

node_modules/o1js/dist/node/lib/mina/actions/offchain-state.d.ts:116:28 - error TS1109: Expression expected.

116 }>(config: Config, options?: {
                               ~

node_modules/o1js/dist/node/lib/mina/actions/offchain-state.d.ts:130:30 - error TS1005: ',' expected.

130     logTotalCapacity?: number;
                                 ~

node_modules/o1js/dist/node/lib/mina/actions/offchain-state.d.ts:140:33 - error TS1005: ',' expected.

140     maxActionsPerUpdate?: number;
                                    ~

node_modules/o1js/dist/node/lib/mina/actions/offchain-state.d.ts:141:32 - error TS1005: ',' expected.

141     maxActionsPerProof?: number;
                                   ~

node_modules/o1js/dist/node/lib/mina/actions/offchain-state.d.ts:142:2 - error TS1005: ';' expected.

142 }): OffchainState<Config>;
     ~

node_modules/o1js/dist/node/lib/mina/actions/offchain-state.d.ts:142:3 - error TS1128: Declaration or statement expected.

142 }): OffchainState<Config>;
      ~


Found 15 errors in the same file, starting at: node_modules/o1js/dist/node/lib/mina/actions/offchain-state.d.ts:114

```

--- a/README.md
+++ b/README.md
@@ -1,1 +1,18 @@
-# utils
+# zkoracle-utils
+
+`zkoracle-utils` is a collection of utilities designed to facilitate the development of applications leveraging zero-knowledge proofs, specifically tailored for Oracle implementations in blockchain technologies. This toolkit aims to simplify the interaction with Oracles, ensuring data integrity and privacy through the use of zero-knowledge proofs.
+
+## Features
+
+- **Transaction URL Generation**: Dynamically generates URLs for viewing transactions on blockchain explorers, supporting different networks and services.
+- **Transaction Processing**: Facilitates the signing, sending, and confirmation of transactions, including smart contract deployments and updates.
+- **File and Key Management**: Offers utilities for checking file existence, storing private keys securely, and initializing keys for various purposes.
+- **Blockchain Interaction**: Simplifies interactions with the blockchain, including fetching account information, deploying smart contracts, and handling events.
+- **Support for Multiple Networks**: Designed to work with both the Mina protocol and a hypothetical Lightnet system, accommodating different blockchain environments.
+
+## Installation
+
+To install `zkoracle-utils`, run the following command in your project directory:
+
+```bash
+npm install zkoracle-utils

--- a/package.json
+++ b/package.json
@@ -1,6 +1,6 @@
 {
   "name": "@zkoracle/utils",
-  "version": "0.0.20",
+  "version": "0.1.20",
   "description": "utils for zkoracle.",
   "author": "ubinix-warun",
   "license": "Apache-2.0",
@@ -16,6 +16,8 @@
   "scripts": {
     "build": "tsc",
     "distclean": "rm -R node_modules build pnpm-lock.yaml ; pnpm install",
+    "cleanbuild": "rm -R node_modules build pnpm-lock.yaml; npm i; npm run build",
+    "fmtpublish": "npm run fmt; npm publish",
     "distclean-pnpm2npm": "rm -R node_modules build pnpm-lock.yaml ; npm install",
     "buildw": "tsc --watch",
     "coverage": "node --experimental-vm-modules node_modules/jest/bin/jest.js --coverage",

--- a/package.json
+++ b/package.json
@@ -37,10 +37,10 @@
     "@typescript-eslint/parser": "^5.62.0",
     "eslint": "^8.56.0",
     "eslint-plugin-o1js": "0.4.0",
-    "jest": "^27.5.1",
+    "jest": "29.2.2",
     "prettier": "^2.8.8",
-    "ts-jest": "^27.1.5",
-    "typescript": "^4.9.5"
+    "ts-jest": "29.2.2",
+    "typescript": "5.5.3"
   },
   "peerDependencies": {
     "o1js": "1.5.0"

--- a/src/Toolkit.ts
+++ b/src/Toolkit.ts
@@ -30,7 +30,7 @@ export class Toolkit {
 
   public static async processTx(
     config: any,
-    sentTx: Mina.Transaction,
+    sentTx: Mina.Transaction<false, false>,
     keys: PrivateKey[],
     tag: string
   ) {
@@ -40,7 +40,7 @@ export class Toolkit {
     /**
      * note: this tx needs to be signed with `tx.sign()`, because `deploy` uses `requireSignature()` under the hood,
      * so one of the account updates in this tx has to be authorized with a signature (vs proof).
-     * this is necessary for the deploy tx because the initial permissions for all account fields are "signature".
+     * this is necessary for the deployment tx because the initial permissions for all account fields are "signature".
      * (but `deploy()` changes some of those permissions to "proof" and adds the verification key that enables proofs.
      * that's why we don't need `tx.sign()` for the later transactions.)
      */
@@ -161,7 +161,7 @@ export class Toolkit {
     // await fs.mkdir("keys");
     if (!(await this.isFileExists(fs, path))) {
       const feePayerPrivateKey = (await Lightnet.acquireKeyPair()).privateKey;
-      const feePayerAccount = feePayerPrivateKey.toPublicKey();
+      // const feePayerAccount = feePayerPrivateKey.toPublicKey();
 
       await this.storePrivateKey(fs, path, feePayerPrivateKey);
 

